<ui version="4.0" >
 <class>instDialog_q</class>
 <widget class="QDialog" name="instDialog_q" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>875</width>
    <height>619</height>
   </rect>
  </property>
  <property name="sizePolicy" >
   <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="sizeGripEnabled" >
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout_2" >
   <item row="0" column="0" >
    <widget class="QLabel" name="titleLabel" >
     <property name="sizePolicy" >
      <sizepolicy vsizetype="Fixed" hsizetype="MinimumExpanding" >
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize" >
      <size>
       <width>0</width>
       <height>25</height>
      </size>
     </property>
     <property name="font" >
      <font>
       <family>Sans Serif</family>
       <pointsize>14</pointsize>
       <weight>75</weight>
       <italic>false</italic>
       <bold>true</bold>
      </font>
     </property>
     <property name="text" >
      <string>TextLabel</string>
     </property>
     <property name="alignment" >
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0" >
    <widget class="QStackedWidget" name="stackedWidget" >
     <property name="sizePolicy" >
      <sizepolicy vsizetype="MinimumExpanding" hsizetype="Expanding" >
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex" >
      <number>1</number>
     </property>
     <widget class="QWidget" name="select_firewalls_page" >
      <property name="sizePolicy" >
       <sizepolicy vsizetype="MinimumExpanding" hsizetype="MinimumExpanding" >
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="layoutDirection" >
       <enum>Qt::LeftToRight</enum>
      </property>
      <layout class="QGridLayout" >
       <property name="margin" >
        <number>0</number>
       </property>
       <item row="0" column="0" >
        <layout class="QVBoxLayout" >
         <property name="margin" >
          <number>2</number>
         </property>
         <item>
          <widget class="QFrame" name="frame10" >
           <property name="sizePolicy" >
            <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="frameShape" >
            <enum>QFrame::Box</enum>
           </property>
           <property name="frameShadow" >
            <enum>QFrame::Plain</enum>
           </property>
           <layout class="QHBoxLayout" >
            <property name="margin" >
             <number>11</number>
            </property>
            <item>
             <widget class="QLabel" name="selectInfoLabel" >
              <property name="text" >
               <string>&lt;p align="center">&lt;b>&lt;font size="+2">Select firewalls to compile and install.&lt;/font>&lt;/b>&lt;/p></string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="frame12" >
           <property name="sizePolicy" >
            <sizepolicy vsizetype="MinimumExpanding" hsizetype="MinimumExpanding" >
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="frameShape" >
            <enum>QFrame::Box</enum>
           </property>
           <property name="frameShadow" >
            <enum>QFrame::Plain</enum>
           </property>
           <layout class="QGridLayout" name="gridLayout" >
            <item row="1" column="0" >
             <widget class="QFrame" name="frame14" >
              <property name="frameShape" >
               <enum>QFrame::Box</enum>
              </property>
              <property name="frameShadow" >
               <enum>QFrame::Plain</enum>
              </property>
              <layout class="QGridLayout" >
               <property name="margin" >
                <number>11</number>
               </property>
               <item row="0" column="2" >
                <spacer>
                 <property name="orientation" >
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeType" >
                  <enum>QSizePolicy::Expanding</enum>
                 </property>
                 <property name="sizeHint" stdset="0" >
                  <size>
                   <width>100</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="0" column="0" >
                <widget class="QPushButton" name="pushButton16" >
                 <property name="text" >
                  <string>All</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1" >
                <widget class="QPushButton" name="pushButton17" >
                 <property name="text" >
                  <string>None</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="0" column="0" >
             <widget class="QTreeWidget" name="selectTable" >
              <property name="alternatingRowColors" >
               <bool>true</bool>
              </property>
              <property name="selectionMode" >
               <enum>QAbstractItemView::NoSelection</enum>
              </property>
              <column>
               <property name="text" >
                <string>Firewall</string>
               </property>
              </column>
              <column>
               <property name="text" >
                <string>Compile</string>
               </property>
              </column>
              <column>
               <property name="text" >
                <string>Install</string>
               </property>
              </column>
              <column>
               <property name="text" >
                <string>Last Modified</string>
               </property>
              </column>
              <column>
               <property name="text" >
                <string>Last Compiled</string>
               </property>
              </column>
              <column>
               <property name="text" >
                <string>Last Installed</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="progress_page" >
      <layout class="QHBoxLayout" >
       <property name="margin" >
        <number>0</number>
       </property>
       <item>
        <widget class="QFrame" name="frame18_2" >
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Plain</enum>
         </property>
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>11</number>
          </property>
          <item row="2" column="1" >
           <widget class="QPushButton" name="controlMCButton" >
            <property name="text" >
             <string>Stop</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType" >
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>112</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item rowspan="2" row="0" column="0" colspan="2" >
           <widget class="QTreeWidget" name="fwWorkList" >
            <property name="alternatingRowColors" >
             <bool>true</bool>
            </property>
            <property name="itemsExpandable" >
             <bool>false</bool>
            </property>
            <property name="sortingEnabled" >
             <bool>true</bool>
            </property>
            <column>
             <property name="text" >
              <string>Firewall</string>
             </property>
            </column>
            <column>
             <property name="text" >
              <string>Progress</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frame18" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Preferred" hsizetype="Expanding" >
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Plain</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout_4" >
          <item row="0" column="0" >
           <layout class="QGridLayout" name="gridLayout_3" >
            <item row="0" column="0" >
             <widget class="QLabel" name="textLabel1" >
              <property name="text" >
               <string>Firewalls:</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="0" column="1" >
             <widget class="QLabel" name="fwMCLabel" >
              <property name="sizePolicy" >
               <sizepolicy vsizetype="Preferred" hsizetype="Expanding" >
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font" >
               <font>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text" >
               <string>firewall</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="0" column="2" >
             <widget class="QProgressBar" name="compFirewallProgress" >
              <property name="minimumSize" >
               <size>
                <width>300</width>
                <height>0</height>
               </size>
              </property>
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="2" >
             <widget class="QLabel" name="infoMCLabel" >
              <property name="sizePolicy" >
               <sizepolicy vsizetype="Preferred" hsizetype="Expanding" >
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font" >
               <font>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text" >
               <string>current operation here</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="1" column="2" >
             <widget class="QProgressBar" name="compProgress" >
              <property name="minimumSize" >
               <size>
                <width>300</width>
                <height>0</height>
               </size>
              </property>
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0" >
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QGroupBox" name="detailMCframe" >
              <property name="title" >
               <string>Process log</string>
              </property>
              <layout class="QGridLayout" >
               <property name="margin" >
                <number>0</number>
               </property>
               <item row="0" column="0" >
                <spacer>
                 <property name="orientation" >
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeType" >
                  <enum>QSizePolicy::Expanding</enum>
                 </property>
                 <property name="sizeHint" stdset="0" >
                  <size>
                   <width>131</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="0" column="1" >
                <widget class="QPushButton" name="saveMCLogButton" >
                 <property name="text" >
                  <string>Save log to file</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0" colspan="2" >
                <widget class="QTextBrowser" name="procLogDisplay" >
                 <property name="openLinks" >
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <spacer>
              <property name="orientation" >
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>16</width>
                <height>210</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item row="2" column="0" >
           <widget class="QFrame" name="batchInstFlagFrame" >
            <property name="sizePolicy" >
             <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="frameShape" >
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow" >
             <enum>QFrame::Plain</enum>
            </property>
            <layout class="QGridLayout" name="_2" >
             <property name="margin" >
              <number>0</number>
             </property>
             <item row="1" column="0" >
              <widget class="QCheckBox" name="batchInstall" >
               <property name="sizePolicy" >
                <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text" >
                <string>Perform batch install</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0" >
              <widget class="QLabel" name="textLabel2" >
               <property name="text" >
                <string>Check this option if you want to install all selected firewalls automatically. This only works if you use the same user name and password to authenticate to all these firewalls. </string>
               </property>
               <property name="alignment" >
                <set>Qt::AlignVCenter</set>
               </property>
               <property name="wordWrap" >
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_3" >
      <layout class="QGridLayout" >
       <item row="1" column="0" >
        <widget class="QTreeWidget" name="lastListView" >
         <property name="verticalScrollBarPolicy" >
          <enum>Qt::ScrollBarAlwaysOn</enum>
         </property>
         <property name="indentation" >
          <number>20</number>
         </property>
         <property name="itemsExpandable" >
          <bool>false</bool>
         </property>
         <column>
          <property name="text" >
           <string>Firewall</string>
          </property>
         </column>
         <column>
          <property name="text" >
           <string>Compile</string>
          </property>
         </column>
         <column>
          <property name="text" >
           <string>Install</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="0" >
    <widget class="QFrame" name="frame" >
     <property name="sizePolicy" >
      <sizepolicy vsizetype="Fixed" hsizetype="MinimumExpanding" >
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize" >
      <size>
       <width>400</width>
       <height>50</height>
      </size>
     </property>
     <property name="frameShape" >
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow" >
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" >
      <item>
       <spacer>
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="backButton" >
        <property name="text" >
         <string>&lt; &amp;Back</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="nextButton" >
        <property name="text" >
         <string>&amp;Next ></string>
        </property>
        <property name="flat" >
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="finishButton" >
        <property name="text" >
         <string>&amp;Finish</string>
        </property>
        <property name="flat" >
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="cancelButton" >
        <property name="text" >
         <string>&amp;Cancel</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <resources/>
 <connections>
  <connection>
   <sender>saveMCLogButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>instDialog_q</receiver>
   <slot>saveLog()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>630</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fwWorkList</sender>
   <signal>itemActivated(QTreeWidgetItem*, int)</signal>
   <receiver>instDialog_q</receiver>
   <slot>findFirewallInCompileLog(QTreeWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>40</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton16</sender>
   <signal>clicked(bool)</signal>
   <receiver>instDialog_q</receiver>
   <slot>selectAllFirewalls()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>43</x>
     <y>493</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton17</sender>
   <signal>clicked(bool)</signal>
   <receiver>instDialog_q</receiver>
   <slot>deselectAllFirewalls()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>93</x>
     <y>493</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>selectTable</sender>
   <signal>itemChanged(QTreeWidgetItem*,int)</signal>
   <receiver>instDialog_q</receiver>
   <slot>tableItemChanged(QTreeWidgetItem*,int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>389</x>
     <y>319</y>
    </hint>
    <hint type="destinationlabel" >
     <x>389</x>
     <y>288</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>procLogDisplay</sender>
   <signal>anchorClicked(QUrl)</signal>
   <receiver>instDialog_q</receiver>
   <slot>logItemClicked(QUrl)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>706</x>
     <y>264</y>
    </hint>
    <hint type="destinationlabel" >
     <x>778</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>tableItemChanged(QTreeWidgetItem*,int)</slot>
  <slot>logItemClicked(QUrl)</slot>
 </slots>
</ui>
